### ISSUE TITLE
Add Metadata to an Org in PCF

### ISSUE THREAD

**Issue by marunk20** [cloudfoundry/cf-java-client#1027]
I am using [cf-java-client](https://github.com/cloudfoundry/cf-java-client) to automate few admin-operations in `PCF`

I need to add [Metadata](https://docs.pivotal.io/pivotalcf/2-6/adminguide/metadata.html) to the newly created and existing Orgs in PCF through the java-client

These are the `java-libraries` that i currently use

```
 compile 'org.cloudfoundry:cloudfoundry-client-reactor:3.15.0.RELEASE'
 compile 'org.cloudfoundry:cloudfoundry-operations:3.15.0.RELEASE'
```

I couldn't see any such methods or class in these libraries that supports adding Metadata to the Org. Appreciate any support on this

**Comment by twoseat** [comment]
At the moment `metadata` is marked as [experimental](https://v3-apidocs.cloudfoundry.org/version/3.79.0/index.html#update-an-organization), and we don't generally add support for experimental features. Having said that, if you're able to provide a PR (perhaps using [this](https://github.com/cloudfoundry/cf-java-client/pull/966) as inspiration?) I'll try to prioritize including it.

In the meantime I'll leave this open as a request for the feature. And I'll encourage you to update to a more recent version of the client, a lot has changed behind the scenes that is worth catching up on!

**Comment by marunk20** [comment]
I am able to get the `MetaData` . But am unable to update the `Organization` with the new `MetaData`

**Comment by twoseat** [comment]
What command do you use to get the `metadata`?

**Comment by marunk20** [comment]
```java
private void showMetaData() {
        OrganizationInfoRequest request = OrganizationInfoRequest.builder()
                .name(ORG_NAME).build();
        OrganizationDetail organizationDetail = cfOps.organizations().get(request).block();




        GetOrganizationRequest getOrganizationRequest= GetOrganizationRequest.builder()
                .organizationId(organizationDetail.getId())
                .build();



        GetOrganizationResponse response = cfOps.getCloudFoundryClient().organizationsV3()

                .get(getOrganizationRequest).block();
        var organizationMetadata = response.getMetadata();
        log.info("Metadata from  "+ORG_NAME+" :: "+organizationMetadata.getLabels());

    }
```

**Comment by twoseat** [comment]
Thanks for that @marunk20. This is an interesting case. The `metadata` field is marked as experimental in the docs, and as such is something we wouldn't support. It's also returned by default in recent versions of the cloud controller, which doesn't sound terribly experimental!

We added limited support for it last month purely to keep our integration tests running, but that didn't extend to adding new functionality such as `update`. I'll take a look at adding that soon, but in the meantime I'm afraid there isn't a workaround for this.

**Comment by marunk20** [comment]
Only workaround that I can think of it, at the moment, is to call PCF Endpoints directly withtout using java-client and write the same in java once the update is availabe

**Comment by marunk20** [comment]
Can this be prioritized please ?

**Comment by checketts** [comment]
This issue is around `metadata` for Orgs. I'm needing `metadata` for Spaces. Should I just watch this issue or open a new one?

**Comment by checketts** [comment]
@marunk20 I just created #1031 to address this. Hopefully it is accepted.

**Comment by twoseat** [comment]
Fixed in #1031
